lib/improver/cli/nowcast_extrapolate.py

```diff
@@ -31,126 +31,29 @@
-    speed_cube = direction_cube = None
-
-    input_cube = load_cube(args.input_filepath)
-    orographic_enhancement_cube = load_cube(
-        args.orographic_enhancement_filepaths, allow_none=True)
-
-    s_path, d_path = (args.advection_speed_filepath,
-                      args.advection_direction_filepath)
-    level_constraint = Constraint(pressure=args.pressure_level)
-    if s_path and d_path:
-        try:
-            speed_cube = load_cube(s_path, constraints=level_constraint)
-            direction_cube = load_cube(d_path, constraints=level_constraint)
-        except ValueError as err:
-            raise ValueError(
-                '{} Unable to extract specified pressure level from given '
-                'speed and direction files.'.format(err))
-
-    attributes_dict = load_json_or_none(args.json_file)
-    # Process Cubes
-    result = process(
-        input_cube, u_cube, v_cube, speed_cube, direction_cube,
-        orographic_enhancement_cube, attributes_dict, args.max_lead_time,
-        args.lead_time_interval)
-
-    # Save Cube
-    save_netcdf(result, args.output_filepath)
-
-
-def process(input_cube, u_cube=None, v_cube=None, speed_cube=None,
-            direction_cube=None, orographic_enhancement_cube=None,
-            attributes_dict=None, max_lead_time=360, lead_time_interval=15):
+@cli.clizefy
+@cli.with_output
+def process(input_cube: cli.inputcube,
+            advection_cubes: inputadvection,
+            orographic_enhancement_cube: cli.inputcube = None,
+            *,
+            attributes_dict: cli.inputjson = None,
+            max_lead_time: int = 360, lead_time_interval: int = 15):
     """Module  to extrapolate input cubes given advection velocity fields.
 
     Args:
         input_cube (iris.cube.Cube):
             The input Cube to be processed.
-        u_cube (iris.cube.Cube):
-            Cube with the velocities in the x direction.
-            Must be used with v_cube.
-            speed_cube and direction_cube must be None.
-        v_cube (iris.cube.Cube):
-            Cube with the velocities in the y direction.
-            Must be used with u_cube.
-            speed_cube and direction_cube must be None.
-        speed_cube (iris.cube.Cube):
-            Cube containing advection speeds, usually wind speed.
-            Must be used with direction_cube.
-            u_cube and v_cube must be None.
-        direction_cube (iris.cube.Cube):
-            Cube from which advection speeds are coming. The directions
-            should be on the same grid as the input speeds, including the same
-            vertical levels.
-            Must be used with speed_cube.
-            u_cube and v_cube must be None.
+        advection_cubes (iris.cube.CubeList):
+            Advection cubes of U and V.
         orographic_enhancement_cube (iris.cube.Cube):
             Cube containing the orographic enhancement fields. May have data
             for multiple times in the cube.
```

Comment:
Suggest: `Cube containing orographic enhancement forecasts for the lead times at which an extrapolation nowcast is required.`.  Also don't need `Default is None` type statements in this docstring.

Reply:
Done